packages/protocol/contracts/ACE/noteRegistry/interfaces/Behaviour.sol

```diff
@@ -0,0 +1,83 @@
+pragma solidity >=0.5.0 <0.6.0;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+
+import "../../../interfaces/IAZTEC.sol";
+import "../proxies/BaseAdminUpgradeabilityProxy.sol";
+import "../Manager.sol";
+
+/**
+ * @title NoteRegistryBehaviour contract which contains the storage variables that define the set of valid
+ * AZTEC notes for a particular address
+ * @author AZTEC
+ * @dev TODO
+ * Copyright Spilbury Holdings Ltd 2019. All rights reserved.
+ **/
+contract NoteRegistryBehaviour is Ownable, IAZTEC {
+    using SafeMath for uint256;
+
+    bool public isActiveBehaviour;
+    bool public initialised;
+    address public dataLocation;
+
+    constructor () Ownable() public {
+        isActiveBehaviour = true;
+    }
+
+    function initialise(
```

Comment:
Given that this contract is just an interface, should the code for this `initialise()` function not be moved out? I would put it in `noteRegistry/epochs/201907/Behaviour.sol`, the base behaviour contract from which all other behaviour contracts inherit. 

Then for the interface, just put `function initialise(address _newOwner, etc) public`

Reply:
I'm not against this, but thought I'd rather always guarantee that this initialise function, even if not re-implemented, will at least transferOwnership and set the initialised flag.

It arguably does make it harder to notice that you are not setting your initial NoteRegistry state for example, but I'd expect that to be discovered through tests you would necessarily be writing, whilst the initialised flag is more unlikely to be caught.

I might add the ownership transfer test + initialised flag test to programatic tests and move this logic.

Reply:
Hmm, after reading your explanation I understand why it's like that - we must gurantee sure that the `initialise` method is available when a new behaviour is deployed and that it's not possible to initialise more than once. 

Perhaps, we could have a `baseBehaviour.sol` contract that includes the code for these important methods. The `epochs/201907/Behaviour.sol` could then inherit from `baseBehaviour.sol` rather than the interface. 

The interface would then just be a standard interface without the code in it